app/forms/mentee_application_state_form.rb

```diff
@@ -0,0 +1,21 @@
+class MenteeApplicationStateForm
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations
+
+  attribute :action, :string
```

Comment:
rationale for using a form object here is that what I care about are the actions, not the different statuses. Using the form object here enables me to leverage the form helpers/make minimal changes to how the form currently functions, but enrich it with the information we need coming from the current state.

Reply:
I still don't see a purpose to this at all. It just becomes less explicit imo for what we need to pass in. I would prefer we just pass in the the 2 variables (current state and mentee application state) instead.

Reply:
If I use the `MenteeApplicationState` model, I can't use the form_for helper and define an `:action` on the form, which I think is much more understandable. I think it reads much more cleanly and don't understand the harm you see in this pattern. 
Can you explain what gets obscured for you and why you think this will be confusing?

Reply:
The obscurity is that we have this new object we have to deal with rather than just an MenteeApplicationState instance. Would you be able to show side by side what this would look like w/o the form object (maybe then i'll be able to better understand why this would be valuable?) (i'm fine if you do so in a gist or something)

Reply:
https://gist.github.com/afogel/01b5900dee784f71f70b4f3b6a1f75b1

In this case, I'd be fine moving to leveraging form tags since it's a simple case, though I still think using `form_with` reads more nicely. Note: We can't use `form_with` in the option without the form object because `:action` is not a defined attribute on the application state model. The gist also contains the modifications that would be necessary for the controller to make it work.

In general, however, I think this form object pattern provides strong value when the complexity of the forms grows and/or the form straddles at least two resources.